app/actions.py

```diff
@@ -92,30 +92,172 @@ def performDeleteRange(self, upperRow, upperCol, lowerRow, lowerCol):
         self.getText(upperRow, upperCol, lowerRow, lowerCol)))
       self.redo()
 
+  def resizeBookmarkList(self):
+    """
+    Repeatedly doubles the length self.bookmarks until
+    its length is at least the number of lines in the document.
+
+    Args:
+      None
+
+    Returns:
+      None
+    """
+    while len(self.bookmarks) < len(self.lines):
+      # Double the length of the array
```

Comment:
I'm concerned about having another array that is as large as the document has lines. Some documents I work with have 20000 lines or more. :) 

Maybe I'm different, but I don't tend to use more than a half dozen bookmarks (just enough to work through a problem).

I'm wondering if this PR is thinking in different scales. Like 2000 lines and hundreds of bookmarks(?).

Reply:
Yeah I was kind of thinking about all the cases. Even though most people only use a few, we don't want the program to crash or have bookmarks overwrite other ones silently. I was also trying to make it quick because although it does take more memory, without this list, you would have to search through a list of sets. However, let me try try this other implementation since I don't think it'll pose a major performance issue.